catkit/hardware/iris_ao/iris_ao_controller.py

```diff
@@ -1,106 +1,170 @@
 """Interface for IrisAO segmented deformable mirror controller."""
 
+import ctypes
 import os
-import signal
-import subprocess
 import time
 
 from catkit.interfaces.DeformableMirrorController import DeformableMirrorController
 from catkit.hardware.iris_ao.segmented_dm_command import SegmentedDmCommand
-
 from catkit.hardware.iris_ao import util
 
 
-class IrisAoDmController(DeformableMirrorController):
-    """
-    The low-level IrisAO API is written in C, and this class bridges to the compiled executable 'DM_Control.exe' stored
-    locally on the machine that controls the IrisAO mirror. The basic functionality is that the user creates a command
-    and stores it to an ini file called 'ConfigPTT.ini'. The executable then grabs that ini file and applies the piston,
-    tip, tilt (PTT) values in that file to the hardware.
-
-    The executable 'DM_Control.exe' is controlled by passing strings to it with stdin. E.g.:
+class IrisAOCLib:
+    def __init__(self, dll_filepath):
+        self.dll_filePath = dll_filepath
+        self.dll = None
+        self.positions_initiated = False
 
-    dm.stdin.write('config\n')
-    dm.stdin.flush()
+        try:
+            self.dll = ctypes.cdll.LoadLibrary(self.dll_filepath)
+        except Exception as error:
+            raise ImportError(f"Failed to load library '{self.dll_filepath}'.") from error
+
+        # Declare prototypes.
+        # void MirrorCommand (void* mirror, void* command);
+        self._MirrorCommand = self.dll.MirrorCommand
+        self._MirrorCommand.restype = None
+        self._MirrorCommand.argtypes = [ctypes.c_void_p, ctypes.c_void_p]
+
+        # void* MirrorConnect (char* mirror_serial, char* driver_serial, bool disabled);
+        self._MirrorConnect = self.dll.MirrorConnect
+        self._MirrorConnect.restype = ctypes.c_void_p
+        self._MirrorConnect.argtypes = [ctypes.c_char_p, ctypes.c_char_p, c_bool]
+
+        # void* MirrorRelease (void* mirror);
+        self._MirrorRelease = self.dll.MirrorConnect
+        self._MirrorRelease.restype = ctypes.c_void_p
+        self._MirrorRelease.argtypes = [ctypes.c_void_p]
+
+    def MirrorCommand(self, mirror):
+        if self.positions_initiated:
+            return self._MirrorConnect(mirror, self.instrument_lib.MirrorSendSettings)
+        else:
+            # TODO: What's the diff?
```

Comment:
It's possible that the difference is that ``MirrorInitSettings`` is the command to apply the .mcf flat. In which case, we might want to expose this, though, I'm not sure of the implications.

All of the C++ examples use ``MirrorInitSettings`` first and only once, and then ``MirrorSendSettings`` for all subsequent calls.

Reply:
Manual says it's ok and lists the following example.
```C++
//This following code example demonstrates how to flatten the mirror segments:
// connect to the specified mirror device
MirrorHandle mirror = MirrorConnect(“FSC37-01-11-0613", “08130002", false);
// flatten all of the mirror segments
MirrorCommand(mirror, MirrorInitSettings);
// release the specified
MirrorRelease(mirror);
```